modules/reconciliation/model/reconciliation.class.php

```diff
@@ -0,0 +1,1120 @@
+          $this->SumAccountH[$AccountPlanID]->FDiff = $Account->FAmountIn - $Account->FAmountOut;
+        }
+
+        $this->total['total']->Diff = $this->total['total']->AmountIn - $this->total['total']->AmountOut;
+        $this->total['total']->FDiff = $this->total['total']->FAmountIn - $this->total['total']->FAmountOut;
+        $this->calculateDiff();
+      } else {
+        print "Ingen kontoer er valgt<br>";
+      }
+      $this->markAsUsed();
+    }
+
+    /**
+     * This is to find out if there is a difference between ledger and general ledger sum
+     */
+    function calculateDiff() {
+      $this->total['diff']->Name = 'Differanse';
+      $this->total['diff']->AmountIn =
+        $this->total['account']->AmountIn - $this->total['total']->AmountIn;
+      $this->total['diff']->AmountOut =
+        $this->total['account']->AmountOut - $this->total['total']->AmountOut;
+      $this->total['diff']->FAmountIn =
+        $this->total['account']->FAmountIn - $this->total['total']->FAmountIn;
+      $this->total['diff']->FAmountOut =
+        $this->total['account']->FAmountOut - $this->total['total']->FAmountOut;
+
+      $this->total['diff']->Diff =
+        round($this->total['diff']->AmountIn - $this->total['diff']->AmountOut, 2);
+      $this->total['diff']->FDiff =
+        round($this->total['diff']->FAmountIn - $this->total['diff']->FAmountOut, 2);
+
+      $this->total['account']->Diff =
+        round($this->total['account']->AmountIn - $this->total['account']->AmountOut, 2);
+      $this->total['account']->FDiff =
+        round($this->total['account']->FAmountIn - $this->total['account']->FAmountOut, 2);
+    }
+
+    /**
+     * Goes through the potential matches and if the sum of the match group is 0 and
+     * all of the vouchers for this group are not marked as used, mark them
+     */
+    function markAsUsed() {
+      $MatchOrder = array("InvoiceID", "KID", "MatchNumber");
+
+      // for each account
+      foreach ($this->MatchH as $AccountPlanID => $Matches) {
+        foreach ($MatchOrder as $MatchedBy) {
+          // If nothing set for this group, skip it
+          if (!isset($this->MatchH[$AccountPlanID][$MatchedBy])) {
+            continue;
+          }
+
+          foreach ($Matches[$MatchedBy] as $MatchKey => $MatchKeySum) {
+            // If the sum is set and it is 0
+            if (isset($this->MatchH[$AccountPlanID][$MatchedBy][$MatchKey])
+              && round($MatchKeySum, 2) == 0)
+            {
+              // Check if any of the vouchers were used already and set their highlight link HTML string
+              $Used = false;
+              $HighlightLinkHTML = array();
+              foreach ($Matches[$MatchedBy."Journals"][$MatchKey] as $VoucherListEntry) {
+                $JournalID = $VoucherListEntry['JournalID'];
+                $VoucherID = $VoucherListEntry['VoucherID'];
+                if (in_array($VoucherID, $this->Matched)) {
```

Comment:
Done in all places. We now set the value for each id used to true in the array and have acces speed of O(1) instead of O(n) in the case of in_array() function usage